controllers/database/postgresql.go

```diff
@@ -1,36 +1,73 @@
 package database
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/go-logr/logr"
 	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
+	"github.com/goharbor/harbor-cluster-operator/controllers/k8s"
 	"github.com/goharbor/harbor-cluster-operator/lcm"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
 )
 
 type PostgreSQLReconciler struct {
 	HarborCluster *goharborv1.HarborCluster
+	CXT           context.Context
+	Client        k8s.Client
+	Recorder      record.EventRecorder
+	Log           logr.Logger
+	DClient       k8s.DClient
+	Scheme        *runtime.Scheme
+	ExpectCR      *unstructured.Unstructured
+	ActualCR      *unstructured.Unstructured
+	Labels        map[string]string
+	Name          string
+	Namespace     string
+	CRStatus      *lcm.CRStatus
+	Connect       *Connect
+	Properties    *lcm.Properties
 }
 
 // Reconciler implements the reconcile logic of postgreSQL service
-func (postgresql *PostgreSQLReconciler) Reconcile() (*lcm.CRStatus, error) {
-	// TODO
-	return nil, nil
+func (postgre *PostgreSQLReconciler) Reconcile() (*lcm.CRStatus, error) {
+
+	postgre.Client.WithContext(postgre.CXT)
+	postgre.DClient.WithContext(postgre.CXT)
+
+	crStatus, err := postgre.Provision()
```

Comment:
I think it should check whether  postgresCR is existed. If not, then you need to call `Provision`

Reply:
the `Provision ` will check PostgreSQL has existed. if existed `Provision` will quit reconciling.